#!/bin/bash

RUN_DIR=/var/cache/pyprobe
STATE_FILE=/var/cache/pyprobe/pyprobe.state
CONFIG_DIR=/etc

if [ ! -d "${RUN_DIR}" ]; then
    mkdir -p ${RUN_DIR}
fi

if [ ! -e "${STATE_FILE}" ]; then
	UUID=$(hostname)-$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 6 | head -n 1)
	echo "{\"gid\": \"${UUID}\", \"state_version\": 1}" > ${STATE_FILE}
	
	echo
	echo This agent will use GID ${UUID} to connect to your core server. Please set the 
	echo key to connect in ${CONFIG_DIR}/pyprobe
	echo 
	
else
	echo ${RUN_DIR} already exists. Leaving it untouched. Please check its contents manually.
fi

if [ ! -e "/usr/bin/python-pyprobe" ]; then
	ln -s /usr/share/python/python-pyprobe/bin/pyprobe /usr/bin/python-pyprobe
fi

# Create default config
if [ ! -e "/etc/pyprobe" ]; then
	cat > /etc/pyprobe <<END
[General]
key = <insert your key here>
name = ${HOSTNAME}
interval = 30
host = <url of your core server>
END

	# Set permissions of config file (it contains key to access server)
	chown root:root /etc/pyprobe
	chmod 600 /etc/pyprobe
else
	chown root:root /etc/pyprobe
	chmod 600 /etc/pyprobe
	echo /etc/pyprobe already exsists. Please check its contents before running the probe.
fi

# Even though http://upstart.ubuntu.com/cookbook/#upstart-in-debian-and-ubuntu-debian-and-ubuntu-specific 
# says that the links in /etc/init.d are not required for upstart jobs anymore: If they are not present,
# service python-pyprobe start will not work. So, let's just create the link manually. 
/sbin/initctl reload-configuration
if [ ! -e /etc/init.d/python-pyprobe ]; then	
	ln -s /lib/init/upstart-job /etc/init.d/python-pyprobe
fi